src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java

```diff
@@ -89,6 +90,7 @@ protected FileMonitoringController doLaunch(FilePath workspace, Launcher launche
 
         protected FileMonitoringController(FilePath ws) throws IOException, InterruptedException {
             // can't keep ws reference because Controller is expected to be serializable
+            ws.mkdirs();
```

Comment:
`controlDir(ws)` - just below - is calling `ws.child("."+id);` so the subsequent call to `mkdirs()` would create the `ws` dirs too, so this line is perhaps not necessary (?)